store/postgres/src/relational_queries.rs

```diff
@@ -932,7 +967,114 @@ impl<'a> QueryFilter<'a> {
         QueryFilter {
             filter,
             table: self.table,
+            layout: self.layout,
+            block: self.block,
+            table_prefix: self.table_prefix.clone(),
+        }
+    }
+
+    fn child(
+        &self,
+        attribute: &Attribute,
+        entity_type: &'a EntityType,
+        filter: &'a EntityFilter,
+        derivative: &'a EntityFilterDerivative,
+        mut out: AstPass<Pg>,
+    ) -> QueryResult<()> {
+        let child_table = self
+            .layout
+            .table_for_entity(entity_type)
+            .expect("Table for child entity not found");
+
+        let child_prefix = "i.";
+        let parent_prefix = "c.";
+
+        out.push_sql("exists (select 1");
+
+        out.push_sql(" from ");
+        out.push_sql(child_table.qualified_name.as_str());
+        out.push_sql(" as i");
+
+        out.push_sql(" where ");
+
+        // Join tables
+        if derivative.is_derived() {
+            // If the parent is derived,
+            // the child column is picked based on the provided attribute
+            // and the parent column is the primary key of the parent table
+            let child_column = child_table
+                .column_for_field(attribute)
+                .expect("Column for an attribute not found");
+            let parent_column = self.table.primary_key();
+
+            if child_column.is_list() {
+                // Type A: p.id = any(c.{parent_field})
```

Comment:
Shouldn't that be `c.id = any(i.{parent_field})` ? And similar in the comments for the other cases

Reply:
7305f53a28989758fcf3ce04123940ac559c03f1